docs/providers.rst

```diff
@@ -1269,6 +1269,46 @@ With the default settings, Wikimedia user identities (meta.wikimedia.org) will b
 App registration for Wikimedia wikis:
     https://meta.wikimedia.org/wiki/Special:OAuthConsumerRegistration/propose
 
+Metamask
+---------
+
+To use Metamask, you must install web3 via "pip install web3" to your project, or put it in your requirements file.
+
+This module implements an ajax powered login flow between metamask and django-allauth.
+
+The javascript first checks for Metamask, and if it is available then it grabs the wallet address. django-allauth uses this wallet address as the username. Be aware that every wallet you connect to the site will generate a new account. You can have only one account attached to the site at a time, but to log in as another account you must disconnect Metamask as you log out from django, and reconnect using a different wallet address.
```

Comment:
> every wallet you connect to the site will generate a new account.

Users can connect multiple Google accounts to their site account -- why wouldn't they be able to do the same with this provider?

Reply:
What will have to happen is there will have to be a javascript function on every page checking the selected Metamask address. The thing is there isn't a central authority or any kind of callback to tell how many addresses are in your metamask wallet, you can only check for the changes to selectedAddress. I'll improve this functionality tonight.

Reply:
Users can do this. It was a misunderstanding on how this works. All manipulation is now done on socialaccount.uid.